File: clients/algoliasearch-client-php/lib/Model/Search/GetLogsResponseLogs.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * GetLogsResponseLogs Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class GetLogsResponseLogs extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'timestamp' => 'string',
        'method' => 'string',
        'answerCode' => 'string',
        'queryBody' => 'string',
        'answer' => 'string',
        'url' => 'string',
        'ip' => 'string',
        'queryHeaders' => 'string',
        'sha1' => 'string',
        'nbApiCalls' => 'string',
        'processingTimeMs' => 'string',
        'index' => 'string',
        'queryParams' => 'string',
        'queryNbHits' => 'string',
        'innerQueries' => '\Algolia\AlgoliaSearch\Model\Search\GetLogsResponseInnerQueries[]',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'timestamp' => null,
        'method' => null,
        'answerCode' => null,
        'queryBody' => null,
        'answer' => null,
        'url' => null,
        'ip' => null,
        'queryHeaders' => null,
        'sha1' => null,
        'nbApiCalls' => null,
        'processingTimeMs' => null,
        'index' => null,
        'queryParams' => null,
        'queryNbHits' => null,
        'innerQueries' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'timestamp' => 'setTimestamp',
        'method' => 'setMethod',
        'answerCode' => 'setAnswerCode',
        'queryBody' => 'setQueryBody',
        'answer' => 'setAnswer',
        'url' => 'setUrl',
        'ip' => 'setIp',
        'queryHeaders' => 'setQueryHeaders',
        'sha1' => 'setSha1',
        'nbApiCalls' => 'setNbApiCalls',
        'processingTimeMs' => 'setProcessingTimeMs',
        'index' => 'setIndex',
        'queryParams' => 'setQueryParams',
        'queryNbHits' => 'setQueryNbHits',
        'innerQueries' => 'setInnerQueries',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'timestamp' => 'getTimestamp',
        'method' => 'getMethod',
        'answerCode' => 'getAnswerCode',
        'queryBody' => 'getQueryBody',
        'answer' => 'getAnswer',
        'url' => 'getUrl',
        'ip' => 'getIp',
        'queryHeaders' => 'getQueryHeaders',
        'sha1' => 'getSha1',
        'nbApiCalls' => 'getNbApiCalls',
        'processingTimeMs' => 'getProcessingTimeMs',
        'index' => 'getIndex',
        'queryParams' => 'getQueryParams',
        'queryNbHits' => 'getQueryNbHits',
        'innerQueries' => 'getInnerQueries',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['timestamp'])) {
            $this->container['timestamp'] = $data['timestamp'];
        }
        if (isset($data['method'])) {
            $this->container['method'] = $data['method'];
        }
        if (isset($data['answerCode'])) {
            $this->container['answerCode'] = $data['answerCode'];
        }
        if (isset($data['queryBody'])) {
            $this->container['queryBody'] = $data['queryBody'];
        }
        if (isset($data['answer'])) {
            $this->container['answer'] = $data['answer'];
        }
        if (isset($data['url'])) {
            $this->container['url'] = $data['url'];
        }
        if (isset($data['ip'])) {
            $this->container['ip'] = $data['ip'];
        }
        if (isset($data['queryHeaders'])) {
            $this->container['queryHeaders'] = $data['queryHeaders'];
        }
        if (isset($data['sha1'])) {
            $this->container['sha1'] = $data['sha1'];
        }
        if (isset($data['nbApiCalls'])) {
            $this->container['nbApiCalls'] = $data['nbApiCalls'];
        }
        if (isset($data['processingTimeMs'])) {
            $this->container['processingTimeMs'] = $data['processingTimeMs'];
        }
        if (isset($data['index'])) {
            $this->container['index'] = $data['index'];
        }
        if (isset($data['queryParams'])) {
            $this->container['queryParams'] = $data['queryParams'];
        }
        if (isset($data['queryNbHits'])) {
            $this->container['queryNbHits'] = $data['queryNbHits'];
        }
        if (isset($data['innerQueries'])) {
            $this->container['innerQueries'] = $data['innerQueries'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['timestamp']) || $this->container['timestamp'] === null) {
            $invalidProperties[] = "'timestamp' can't be null";
        }
        if (!isset($this->container['method']) || $this->container['method'] === null) {
            $invalidProperties[] = "'method' can't be null";
        }
        if (!isset($this->container['answerCode']) || $this->container['answerCode'] === null) {
            $invalidProperties[] = "'answerCode' can't be null";
        }
        if (!isset($this->container['queryBody']) || $this->container['queryBody'] === null) {
            $invalidProperties[] = "'queryBody' can't be null";
        }
        if (!isset($this->container['answer']) || $this->container['answer'] === null) {
            $invalidProperties[] = "'answer' can't be null";
        }
        if (!isset($this->container['url']) || $this->container['url'] === null) {
            $invalidProperties[] = "'url' can't be null";
        }
        if (!isset($this->container['ip']) || $this->container['ip'] === null) {
            $invalidProperties[] = "'ip' can't be null";
        }
        if (!isset($this->container['queryHeaders']) || $this->container['queryHeaders'] === null) {
            $invalidProperties[] = "'queryHeaders' can't be null";
        }
        if (!isset($this->container['sha1']) || $this->container['sha1'] === null) {
            $invalidProperties[] = "'sha1' can't be null";
        }
        if (!isset($this->container['nbApiCalls']) || $this->container['nbApiCalls'] === null) {
            $invalidProperties[] = "'nbApiCalls' can't be null";
        }
        if (!isset($this->container['processingTimeMs']) || $this->container['processingTimeMs'] === null) {
            $invalidProperties[] = "'processingTimeMs' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets timestamp
     *
     * @return string
     */
    public function getTimestamp()
    {
        return $this->container['timestamp'] ?? null;
    }

    /**
     * Sets timestamp
     *
     * @param string $timestamp timestamp in ISO-8601 format
     *
     * @return self
     */
    public function setTimestamp($timestamp)
    {
        $this->container['timestamp'] = $timestamp;

        return $this;
    }

    /**
     * Gets method
     *
     * @return string
     */
    public function getMethod()
    {
        return $this->container['method'] ?? null;
    }

    /**
     * Sets method
     *
     * @param string $method HTTP method of the perfomed request
     *
     * @return self
     */
    public function setMethod($method)
    {
        $this->container['method'] = $method;

        return $this;
    }

    /**
     * Gets answerCode
     *
     * @return string
     */
    public function getAnswerCode()
    {
        return $this->container['answerCode'] ?? null;
    }

    /**
     * Sets answerCode
     *
     * @param string $answerCode HTTP response code
     *
     * @return self
     */
    public function setAnswerCode($answerCode)
    {
        $this->container['answerCode'] = $answerCode;

        return $this;
    }

    /**
     * Gets queryBody
     *
     * @return string
     */
    public function getQueryBody()
    {
        return $this->container['queryBody'] ?? null;
    }

    /**
     * Sets queryBody
     *
     * @param string $queryBody Request body. Truncated after 1000 characters.
     *
     * @return self
     */
    public function setQueryBody($queryBody)
    {
        $this->container['queryBody'] = $queryBody;

        return $this;
    }

    /**
     * Gets answer
     *
     * @return string
     */
    public function getAnswer()
    {
        return $this->container['answer'] ?? null;
    }

    /**
     * Sets answer
     *
     * @param string $answer Answer body. Truncated after 1000 characters.
     *
     * @return self
     */
    public function setAnswer($answer)
    {
        $this->container['answer'] = $answer;

        return $this;
    }

    /**
     * Gets url
     *
     * @return string
     */
    public function getUrl()
    {
        return $this->container['url'] ?? null;
    }

    /**
     * Sets url
     *
     * @param string $url request URL
     *
     * @return self
     */
    public function setUrl($url)
    {
        $this->container['url'] = $url;

        return $this;
    }

    /**
     * Gets ip
     *
     * @return string
     */
    public function getIp()
    {
        return $this->container['ip'] ?? null;
    }

    /**
     * Sets ip
     *
     * @param string $ip IP of the client which perfomed the request
     *
     * @return self
     */
    public function setIp($ip)
    {
        $this->container['ip'] = $ip;

        return $this;
    }

    /**
     * Gets queryHeaders
     *
     * @return string
     */
    public function getQueryHeaders()
    {
        return $this->container['queryHeaders'] ?? null;
    }

    /**
     * Sets queryHeaders
     *
     * @param string $queryHeaders request Headers (API Key is obfuscated)
     *
     * @return self
     */
    public function setQueryHeaders($queryHeaders)
    {
        $this->container['queryHeaders'] = $queryHeaders;

        return $this;
    }

    /**
     * Gets sha1
     *
     * @return string
     */
    public function getSha1()
    {
        return $this->container['sha1'] ?? null;
    }

    /**
     * Sets sha1
     *
     * @param string $sha1 SHA1 signature of the log entry
     *
     * @return self
     */
    public function setSha1($sha1)
    {
        $this->container['sha1'] = $sha1;

        return $this;
    }

    /**
     * Gets nbApiCalls
     *
     * @return string
     */
    public function getNbApiCalls()
    {
        return $this->container['nbApiCalls'] ?? null;
    }

    /**
     * Sets nbApiCalls
     *
     * @param string $nbApiCalls number of API calls
     *
     * @return self
     */
    public function setNbApiCalls($nbApiCalls)
    {
        $this->container['nbApiCalls'] = $nbApiCalls;

        return $this;
    }

    /**
     * Gets processingTimeMs
     *
     * @return string
     */
    public function getProcessingTimeMs()
    {
        return $this->container['processingTimeMs'] ?? null;
    }

    /**
     * Sets processingTimeMs
     *
     * @param string $processingTimeMs Processing time for the query. It doesn't include network time.
     *
     * @return self
     */
    public function setProcessingTimeMs($processingTimeMs)
    {
        $this->container['processingTimeMs'] = $processingTimeMs;

        return $this;
    }

    /**
     * Gets index
     *
     * @return string|null
     */
    public function getIndex()
    {
        return $this->container['index'] ?? null;
    }

    /**
     * Sets index
     *
     * @param string|null $index index targeted by the query
     *
     * @return self
     */
    public function setIndex($index)
    {
        $this->container['index'] = $index;

        return $this;
    }

    /**
     * Gets queryParams
     *
     * @return string|null
     */
    public function getQueryParams()
    {
        return $this->container['queryParams'] ?? null;
    }

    /**
     * Sets queryParams
     *
     * @param string|null $queryParams query parameters sent with the request
     *
     * @return self
     */
    public function setQueryParams($queryParams)
    {
        $this->container['queryParams'] = $queryParams;

        return $this;
    }

    /**
     * Gets queryNbHits
     *
     * @return string|null
     */
    public function getQueryNbHits()
    {
        return $this->container['queryNbHits'] ?? null;
    }

    /**
     * Sets queryNbHits
     *
     * @param string|null $queryNbHits number of hits returned for the query
     *
     * @return self
     */
    public function setQueryNbHits($queryNbHits)
    {
        $this->container['queryNbHits'] = $queryNbHits;

        return $this;
    }

    /**
     * Gets innerQueries
     *
     * @return \Algolia\AlgoliaSearch\Model\Search\GetLogsResponseInnerQueries[]|null
     */
    public function getInnerQueries()
    {
        return $this->container['innerQueries'] ?? null;
    }

    /**
     * Sets innerQueries
     *
     * @param \Algolia\AlgoliaSearch\Model\Search\GetLogsResponseInnerQueries[]|null $innerQueries array of all performed queries for the given request
     *
     * @return self
     */
    public function setInnerQueries($innerQueries)
    {
        $this->container['innerQueries'] = $innerQueries;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/search/GetLogsResponseLogs.java
package com.algolia.model.search;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/** GetLogsResponseLogs */
public class GetLogsResponseLogs {

  @SerializedName("timestamp")
  private String timestamp;

  @SerializedName("method")
  private String method;

  @SerializedName("answer_code")
  private String answerCode;

  @SerializedName("query_body")
  private String queryBody;

  @SerializedName("answer")
  private String answer;

  @SerializedName("url")
  private String url;

  @SerializedName("ip")
  private String ip;

  @SerializedName("query_headers")
  private String queryHeaders;

  @SerializedName("sha1")
  private String sha1;

  @SerializedName("nb_api_calls")
  private String nbApiCalls;

  @SerializedName("processing_time_ms")
  private String processingTimeMs;

  @SerializedName("index")
  private String index;

  @SerializedName("query_params")
  private String queryParams;

  @SerializedName("query_nb_hits")
  private String queryNbHits;

  @SerializedName("inner_queries")
  private List<GetLogsResponseInnerQueries> innerQueries = null;

  public GetLogsResponseLogs setTimestamp(String timestamp) {
    this.timestamp = timestamp;
    return this;
  }

  /**
   * Timestamp in ISO-8601 format.
   *
   * @return timestamp
   */
  @javax.annotation.Nonnull
  public String getTimestamp() {
    return timestamp;
  }

  public GetLogsResponseLogs setMethod(String method) {
    this.method = method;
    return this;
  }

  /**
   * HTTP method of the perfomed request.
   *
   * @return method
   */
  @javax.annotation.Nonnull
  public String getMethod() {
    return method;
  }

  public GetLogsResponseLogs setAnswerCode(String answerCode) {
    this.answerCode = answerCode;
    return this;
  }

  /**
   * HTTP response code.
   *
   * @return answerCode
   */
  @javax.annotation.Nonnull
  public String getAnswerCode() {
    return answerCode;
  }

  public GetLogsResponseLogs setQueryBody(String queryBody) {
    this.queryBody = queryBody;
    return this;
  }

  /**
   * Request body. Truncated after 1000 characters.
   *
   * @return queryBody
   */
  @javax.annotation.Nonnull
  public String getQueryBody() {
    return queryBody;
  }

  public GetLogsResponseLogs setAnswer(String answer) {
    this.answer = answer;
    return this;
  }

  /**
   * Answer body. Truncated after 1000 characters.
   *
   * @return answer
   */
  @javax.annotation.Nonnull
  public String getAnswer() {
    return answer;
  }

  public GetLogsResponseLogs setUrl(String url) {
    this.url = url;
    return this;
  }

  /**
   * Request URL.
   *
   * @return url
   */
  @javax.annotation.Nonnull
  public String getUrl() {
    return url;
  }

  public GetLogsResponseLogs setIp(String ip) {
    this.ip = ip;
    return this;
  }

  /**
   * IP of the client which perfomed the request.
   *
   * @return ip
   */
  @javax.annotation.Nonnull
  public String getIp() {
    return ip;
  }

  public GetLogsResponseLogs setQueryHeaders(String queryHeaders) {
    this.queryHeaders = queryHeaders;
    return this;
  }

  /**
   * Request Headers (API Key is obfuscated).
   *
   * @return queryHeaders
   */
  @javax.annotation.Nonnull
  public String getQueryHeaders() {
    return queryHeaders;
  }

  public GetLogsResponseLogs setSha1(String sha1) {
    this.sha1 = sha1;
    return this;
  }

  /**
   * SHA1 signature of the log entry.
   *
   * @return sha1
   */
  @javax.annotation.Nonnull
  public String getSha1() {
    return sha1;
  }

  public GetLogsResponseLogs setNbApiCalls(String nbApiCalls) {
    this.nbApiCalls = nbApiCalls;
    return this;
  }

  /**
   * Number of API calls.
   *
   * @return nbApiCalls
   */
  @javax.annotation.Nonnull
  public String getNbApiCalls() {
    return nbApiCalls;
  }

  public GetLogsResponseLogs setProcessingTimeMs(String processingTimeMs) {
    this.processingTimeMs = processingTimeMs;
    return this;
  }

  /**
   * Processing time for the query. It doesn't include network time.
   *
   * @return processingTimeMs
   */
  @javax.annotation.Nonnull
  public String getProcessingTimeMs() {
    return processingTimeMs;
  }

  public GetLogsResponseLogs setIndex(String index) {
    this.index = index;
    return this;
  }

  /**
   * Index targeted by the query.
   *
   * @return index
   */
  @javax.annotation.Nullable
  public String getIndex() {
    return index;
  }

  public GetLogsResponseLogs setQueryParams(String queryParams) {
    this.queryParams = queryParams;
    return this;
  }

  /**
   * Query parameters sent with the request.
   *
   * @return queryParams
   */
  @javax.annotation.Nullable
  public String getQueryParams() {
    return queryParams;
  }

  public GetLogsResponseLogs setQueryNbHits(String queryNbHits) {
    this.queryNbHits = queryNbHits;
    return this;
  }

  /**
   * Number of hits returned for the query.
   *
   * @return queryNbHits
   */
  @javax.annotation.Nullable
  public String getQueryNbHits() {
    return queryNbHits;
  }

  public GetLogsResponseLogs setInnerQueries(
    List<GetLogsResponseInnerQueries> innerQueries
  ) {
    this.innerQueries = innerQueries;
    return this;
  }

  public GetLogsResponseLogs addInnerQueriesItem(
    GetLogsResponseInnerQueries innerQueriesItem
  ) {
    if (this.innerQueries == null) {
      this.innerQueries = new ArrayList<>();
    }
    this.innerQueries.add(innerQueriesItem);
    return this;
  }

  /**
   * Array of all performed queries for the given request.
   *
   * @return innerQueries
   */
  @javax.annotation.Nullable
  public List<GetLogsResponseInnerQueries> getInnerQueries() {
    return innerQueries;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GetLogsResponseLogs getLogsResponseLogs = (GetLogsResponseLogs) o;
    return (
      Objects.equals(this.timestamp, getLogsResponseLogs.timestamp) &&
      Objects.equals(this.method, getLogsResponseLogs.method) &&
      Objects.equals(this.answerCode, getLogsResponseLogs.answerCode) &&
      Objects.equals(this.queryBody, getLogsResponseLogs.queryBody) &&
      Objects.equals(this.answer, getLogsResponseLogs.answer) &&
      Objects.equals(this.url, getLogsResponseLogs.url) &&
      Objects.equals(this.ip, getLogsResponseLogs.ip) &&
      Objects.equals(this.queryHeaders, getLogsResponseLogs.queryHeaders) &&
      Objects.equals(this.sha1, getLogsResponseLogs.sha1) &&
      Objects.equals(this.nbApiCalls, getLogsResponseLogs.nbApiCalls) &&
      Objects.equals(
        this.processingTimeMs,
        getLogsResponseLogs.processingTimeMs
      ) &&
      Objects.equals(this.index, getLogsResponseLogs.index) &&
      Objects.equals(this.queryParams, getLogsResponseLogs.queryParams) &&
      Objects.equals(this.queryNbHits, getLogsResponseLogs.queryNbHits) &&
      Objects.equals(this.innerQueries, getLogsResponseLogs.innerQueries)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      timestamp,
      method,
      answerCode,
      queryBody,
      answer,
      url,
      ip,
      queryHeaders,
      sha1,
      nbApiCalls,
      processingTimeMs,
      index,
      queryParams,
      queryNbHits,
      innerQueries
    );
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GetLogsResponseLogs {\n");
    sb
      .append("    timestamp: ")
      .append(toIndentedString(timestamp))
      .append("\n");
    sb.append("    method: ").append(toIndentedString(method)).append("\n");
    sb
      .append("    answerCode: ")
      .append(toIndentedString(answerCode))
      .append("\n");
    sb
      .append("    queryBody: ")
      .append(toIndentedString(queryBody))
      .append("\n");
    sb.append("    answer: ").append(toIndentedString(answer)).append("\n");
    sb.append("    url: ").append(toIndentedString(url)).append("\n");
    sb.append("    ip: ").append(toIndentedString(ip)).append("\n");
    sb
      .append("    queryHeaders: ")
      .append(toIndentedString(queryHeaders))
      .append("\n");
    sb.append("    sha1: ").append(toIndentedString(sha1)).append("\n");
    sb
      .append("    nbApiCalls: ")
      .append(toIndentedString(nbApiCalls))
      .append("\n");
    sb
      .append("    processingTimeMs: ")
      .append(toIndentedString(processingTimeMs))
      .append("\n");
    sb.append("    index: ").append(toIndentedString(index)).append("\n");
    sb
      .append("    queryParams: ")
      .append(toIndentedString(queryParams))
      .append("\n");
    sb
      .append("    queryNbHits: ")
      .append(toIndentedString(queryNbHits))
      .append("\n");
    sb
      .append("    innerQueries: ")
      .append(toIndentedString(innerQueries))
      .append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
